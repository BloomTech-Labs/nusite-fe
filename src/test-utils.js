import React from "react";
import { render } from "@testing-library/react";
import { ThemeContext } from "./context/contexts";
import UserContext, { ANONYMOUS_USER } from "./context/user/context";

function MutationObserver(callback) {
   this.observe = jest.fn();
   this.disconnect = jest.fn();
   this.takeRecords = jest.fn();
}
global.MutationObserver = jest.fn(MutationObserver);

const mockSpies = {
   setDarkMode: jest.fn(),
   userDispatch: jest.fn(),
};

const AllTheProviders = ({ children }) => (
   <ThemeContext.Provider
      value={{ darkMode: false, setDarkMode: mockSpies.setDarkMode }}
   >
      <UserContext.Provider
         value={{
            userData: ANONYMOUS_USER,
            userDispatch: mockSpies.userDispatch,
         }}
      >
         {children}
      </UserContext.Provider>
   </ThemeContext.Provider>
);

const customRender = (ui, options) => {
   return {
      ...render(ui, { wrapper: AllTheProviders, ...options }),
      mockSpies,
   };
};

const testLinkArray = (linkArray, expected_length, expected_refs) => {
   expect(linkArray.length).toBe(expected_length);
   linkArray.forEach(element => {
      expect(expected_refs).toContain(element.getAttribute("href"));
   });
};

export * from "@testing-library/react";
export { customRender as render, testLinkArray };
