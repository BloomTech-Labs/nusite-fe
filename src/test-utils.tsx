import React from "react";
import { render, RenderResult } from "@testing-library/react";

import { ThemeContext } from "./context/contexts";

interface SpiesMap {
   setDarkMode: jest.Mock;
}

interface RenderWithSpiesResult extends RenderResult {
   mockSpies: SpiesMap;
}

const mockSpies: SpiesMap = {
   setDarkMode: jest.fn(),
};

const AllTheProviders: React.FC = ({ children }) => (
   <ThemeContext.Provider
      value={{ darkMode: false, setDarkMode: mockSpies.setDarkMode }}
   >
      {children}
   </ThemeContext.Provider>
);

const customRender = (ui: React.ReactElement, options?: Object | undefined) => {
   return {
      ...render(ui, { wrapper: AllTheProviders, ...options }),
      mockSpies,
   };
};

export * from "@testing-library/react";
export { customRender as render, RenderWithSpiesResult };
