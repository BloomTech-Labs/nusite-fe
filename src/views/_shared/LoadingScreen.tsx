import LinearProgress from "@material-ui/core/LinearProgress";
import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
   root: {
      height: 12,
   },
}));

const LoadingScreen = (props: any) => {
   const classes = useStyles();
   return (
      <div>
         <LinearProgress
            className={classes.root}
            aria-busy="true"
         ></LinearProgress>
      </div>
   );
};

export default LoadingScreen;
