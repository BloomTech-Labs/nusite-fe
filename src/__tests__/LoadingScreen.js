import "@testing-library/jest-dom/extend-expect";

import React from "react";
import { render } from "../test-utils";

import LoadingScreen from "../views/_shared/LoadingScreen";

describe("Loading Screen test", () => {
   test("It displays a loading screen correctly without error.", () => {
      const loadingScreen = render(<LoadingScreen />);
      expect(loadingScreen).toMatchSnapshot();
   });
});
