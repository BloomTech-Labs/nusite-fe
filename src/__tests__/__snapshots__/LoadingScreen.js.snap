// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Screen test It displays a loading screen correctly without error. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          aria-busy="true"
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-root-1 MuiLinearProgress-indeterminate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          />
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        aria-busy="true"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary makeStyles-root-1 MuiLinearProgress-indeterminate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        />
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "mockSpies": Object {
    "setDarkMode": [MockFunction],
    "userDispatch": [MockFunction],
  },
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
